src/naive/time/mod.rs

```diff
@@ -190,6 +190,15 @@ pub struct NaiveTime {
     frac: u32,
 }
 
+#[cfg(feature = "arbitrary")]
+impl arbitrary::Arbitrary<'_> for NaiveTime {
+    fn arbitrary(u: &mut arbitrary::Unstructured) -> arbitrary::Result<NaiveTime> {
+        let secs = u.int_in_range(0..=86_400)?;
+        let frac = u.int_in_range(0..=2_000_000_000)?;
+        Ok(NaiveTime { secs, frac })
```

Comment:
This feels like something that might be better to use `from_num_seconds_from_midnight_opt` (and unwrap it with a clear error that it is a `chrono` bug if it fails). That way you can't silently make invalid `NaiveTime`s.

And yeah, a secs of 86_400 *will* be rejected.